### PR TITLE
refactor some code to inertiatrait and build pagination links method

### PR DESCRIPTION
**Pull request by ACampanario**

### CODE REVIEW

File: src/Traits/InertiaResponseTrait.php [hunk 1]
```diff
@@ -30,10 +33,40 @@ trait InertiaResponseTrait
...
+                }
+            }
+        }
+
+        $db = ConnectionManager::get('default');
+        $collection = $db->getSchemaCollection();
+        $tableSchema = $collection->describe($model);
+        foreach ($tableSchema->columns() as $column) {
+            if ($tableSchema->getColumnType($column) == 'timestampfractional' && $data[$column] !== null) {
+                $data[$column] = Date::parseDate($data[$column], 'YYYY-MM-dd');
```
Comment (steinkel) [inline src/Traits/InertiaResponseTrait.php]: formatting could be different in specific projects?

File: src/Traits/InertiaResponseTrait.php [hunk 2]
```diff
@@ -30,10 +33,40 @@ trait InertiaResponseTrait
...
+                        foreach ($item as $value) {
+                            $newItems[] = $value['id'];
+                        }
+                    }
+                    $data[$key]['_ids'] = $newItems;
+                }
+            }
+        }
+
+        $db = ConnectionManager::get('default');
```
Comment (steinkel) [inline src/Traits/InertiaResponseTrait.php]: what if the related table is in another datasource?

**Review by steinkel — CHANGES_REQUESTED** [review]

File: src/Traits/InertiaResponseTrait.php (hunk 1 above)
Comment (ACampanario) [reply src/Traits/InertiaResponseTrait.php]: could it be true, I pass the format as a parameter?

File: src/Traits/InertiaResponseTrait.php (hunk 2 above)
Comment (ACampanario) [reply src/Traits/InertiaResponseTrait.php]: could it be true, I pass the datasource as a parameter? then in parseRelated function we would have 
- data
- model
- source

**Pull request merged**